File: KELAS/Zodiak/zodiak.cpp
#include <iostream>
using namespace std;

int main() 
{
    int tgl,bln;
    cout << "PROGRAM ZODIAK ANDA! \n" << endl;
    cin>>bln;
    cout<<"Inputkan Tanggal Lahir Anda: ";
    cin>>tgl;

    if (bln>0&&bln<=12)
    {
        if (bln==1)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Capricorn";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Aquarius \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }
 else
   cout<<"Bln yang anda input tidak valid";
    if (bln>0&&bln<=12)
    {
        if (bln==2)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Aquarius";
            }
        else if (tgl>=20 && tgl<=28)
            {
                cout<<"Bintang Anda Pisces \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==3)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Pisces";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Aries \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==4)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Aries";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Taurus \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==5)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Taurus";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Gemini \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==6)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Gemini";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Cancer \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==7)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Cancer";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Leo \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==8)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Leo";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Virgo \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==9)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Virgo";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Libra \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==10)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Libra";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Scorpio \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==11)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Scorpio";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Sagitarius \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    if (bln>0&&bln<=12)
    {
        if (bln==12)
        {
            if (tgl>=1 &&tgl<=19)
            {
                cout<<"Bintang Anda Sagitarius";
            }
        else if (tgl>=20 && tgl<=31)
            {
                cout<<"Bintang Anda Capricorn \n";
            }
           else {
                cout<<"Tanggal tidak Valid";
                }
        }
    }

    
   return 0;
}

File: TUGAS PAK TRI/12-1-2026/27-01-2026 zodiak/src/zodiak.java
import java.util.Scanner;

/**
 *
 * @author Akbar
 */
public class zodiak {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        System.out.print("Masukkan tanggal lahir (1-31): ");
        int tanggal = input.nextInt();

        System.out.print("Masukkan bulan lahir (1-12): ");
        int bulan = input.nextInt();

        String zodiak;

        if (bulan == 1) {
            if (tanggal <= 20) {
                zodiak = "Capricorn";
            } else {
                zodiak = "Aquarius";
            }

        } else if (bulan == 2) {
            if (tanggal <= 20) {
                zodiak = "Aquarius";
            } else {
                zodiak = "Pisces";
            }

        } else if (bulan == 3) {
            if (tanggal <= 20) {
                zodiak = "Pisces";
            } else {
                zodiak = "Aries";
            }

        } else if (bulan == 4) {
            if (tanggal <= 20) {
                zodiak = "Aries";
            } else {
                zodiak = "Taurus";
            }

        } else if (bulan == 5) {
            if (tanggal <= 20) {
                zodiak = "Taurus";
            } else {
                zodiak = "Gemini";
            }

        } else if (bulan == 6) {
            if (tanggal <= 20) {
                zodiak = "Gemini";
            } else {
                zodiak = "Cancer";
            }

        } else if (bulan == 7) {
            if (tanggal <= 20) {
                zodiak = "Cancer";
            } else {
                zodiak = "Leo";
            }

        } else if (bulan == 8) {
            if (tanggal <= 20) {
                zodiak = "Leo";
            } else {
                zodiak = "Virgo";
            }

        } else if (bulan == 9) {
            if (tanggal <= 20) {
                zodiak = "Virgo";
            } else {
                zodiak = "Libra";
            }

        } else if (bulan == 10) {
            if (tanggal <= 20) {
                zodiak = "Libra";
            } else {
                zodiak = "Scorpio";
            }

        } else if (bulan == 11) {
            if (tanggal <= 20) {
                zodiak = "Scorpio";
            } else {
                zodiak = "Sagittarius";
            }

        } else if (bulan == 12) {
            if (tanggal <= 20) {
                zodiak = "Sagittarius";
            } else {
                zodiak = "Capricorn";
            }

        } else {
            zodiak = "Bulan tidak valid";
        }

        System.out.println("Zodiak Anda adalah: " + zodiak);
    }
}
